opslevel/resource_opslevel_secrets.go

```diff
@@ -131,6 +131,10 @@ func (r *SecretResource) Read(ctx context.Context, req resource.ReadRequest, res
 
 	secret, err := r.client.GetSecret(data.Id.ValueString())
 	if err != nil {
+		if (secret == nil || secret.ID == "") && opslevel.IsOpsLevelApiError(err) {
```

Comment:
This one feels scary - we need to ensure this one is something we want to correct.

Reply:
How about adding a warning here like `resp.Diagnostics.AddWarning("State drift", fmt.Sprintf("Secret with ID '%s' in Terraform state not found in OpsLevel", data.Id.ValueString()))` ?

Reply:
Ok so thinking on it more i think its correct.  If the secret is not found by the ID then the worst it does is repush the secret to us.  Its not actually deleting anything.